### PR TITLE
♻️Sim4Life computational jobs are not parented correctly

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
Issue: the project metadata are not set correctly in the comp_runs table when sim4life runs a job

Observations:
- Tried to reproduce the issue using a Jupyterlab and running the osparc client to no avail, it works there properly,
- Tried to add the exact same metadata and it still works.

Actions:
- added a debug log to see if the call in the director-v2 is doing some weird behavior.
- improved testing in dv-2 regarding metadata

## Related issue/s
- relates to https://github.com/ITISFoundation/osparc-simcore/issues/6530

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Comment by mguidon** [comment]
Just fyi: s4l in prod is still using custom meta data fields for this. Not the headers.

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py [hunk 1]
```diff
@@ -102,13 +102,13 @@ async def create(
...
-                        if cluster_id != DEFAULT_CLUSTER_ID
-                        else None,
+                        cluster_id=(
+                            cluster_id if cluster_id != DEFAULT_CLUSTER_ID else None
+                        ),
                         iteration=iteration,
                         result=RUNNING_STATE_TO_DB[RunningState.PUBLISHED],
-                        started=datetime.datetime.now(tz=datetime.timezone.utc),
-                        metadata=jsonable_encoder(metadata) if metadata else None,
+                        started=datetime.datetime.now(tz=datetime.UTC),
```
Comment (GitHK) [inline services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py:110]: why not use arrow instead like we do in a lot of other palces?

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py [hunk 2]
```diff
@@ -146,7 +146,7 @@ async def set_run_result(
     ) -> CompRunsAtDB | None:
         values: dict[str, Any] = {"result": RUNNING_STATE_TO_DB[result_state]}
         if final_state:
-            values.update({"ended": datetime.datetime.now(tz=datetime.timezone.utc)})
+            values.update({"ended": datetime.datetime.now(tz=datetime.UTC)})
```
Comment (GitHK) [inline services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py:149]: same observation as above

File: services/director-v2/tests/unit/with_dbs/conftest.py [hunk 3]
```diff
@@ -84,28 +89,36 @@ def creator(
...
+                    if node_data.outputs
+                    else {}
+                ),
                 "image": Image(name=node_data.key, tag=node_data.version).dict(  # type: ignore
                     by_alias=True, exclude_unset=True
                 ),  # type: ignore
                 "node_class": to_node_class(node_data.key),
                 "internal_id": internal_id + 1,
-                "submit": datetime.datetime.now(tz=datetime.timezone.utc),
+                "submit": datetime.datetime.now(tz=datetime.UTC),
```
Comment (GitHK) [inline services/director-v2/tests/unit/with_dbs/conftest.py:121]: and also here

**Review by GitHK — APPROVED** [review]

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py (hunk 1 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py:110]: well that is not the purpose of this PR. I am not refactoring this at the moment.

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py (hunk 2 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py:149]: same answer

File: services/director-v2/tests/unit/with_dbs/conftest.py (hunk 3 above)
Comment (sanderegg) [reply services/director-v2/tests/unit/with_dbs/conftest.py:121]: same. this is a RUFF change

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py [hunk 4]
```diff
@@ -102,13 +102,13 @@ async def create(
...
-                        else None,
+                        cluster_id=(
+                            cluster_id if cluster_id != DEFAULT_CLUSTER_ID else None
+                        ),
                         iteration=iteration,
                         result=RUNNING_STATE_TO_DB[RunningState.PUBLISHED],
-                        started=datetime.datetime.now(tz=datetime.timezone.utc),
-                        metadata=jsonable_encoder(metadata) if metadata else None,
+                        started=datetime.datetime.now(tz=datetime.UTC),
+                        metadata=jsonable_encoder(metadata),
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py:111]: `metadata` is nullable. Why do you remove the possibility of setting metadata to null?

**Review by pcrespov — APPROVED** [review]
IF I understand this correctly, you still did not manage to find the cause for the bug and this is just adding some loggers to get more info in the `master` deployment.

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py (hunk 4 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_runs.py:111]: it is not the same metadata in this case

**Comment by sanderegg** [comment]
> IF I understand this correctly, you still did not manage to find the cause for the bug and this is just adding some loggers to get more info in the `master` deployment.

yes

**Pull request merged**